unittests/runtime/Mutex.cpp

```diff
@@ -13,12 +13,57 @@
 #include "swift/Runtime/Mutex.h"
 #include "gtest/gtest.h"
 #include <chrono>
+#include <map>
+#include <random>
 #include <thread>
 
 using namespace swift;
 
+// When true many of the threaded tests log activity to help triage issues.
+static bool trace = false;
+
+// Note the use of non-atomic booleans for "lazy" thread control in many of
+// the tests is sufficient for what we are doing here, no need for
+// atomic operations.
+
+template <typename ThreadBody, typename AfterSpinRelease>
+void threadedExecute(int threadCount, ThreadBody threadBody,
+                     AfterSpinRelease afterSpinRelease) {
+
+  std::vector<std::thread> threads;
+
+  // Block the threads we are about to create.
+  bool spinWait = true;
```

Comment:
In this situation only ever a single writer exists and all readers are looping over the value. You will either get the old value, the new value, or potentially a partial value (don't know if the last one is possible on any supported platforms for a simple bool, even in the partial value case things will function as expected). Also the optimizer should not be able to eliminate the loads so it will get the new value at some point. Is their some other issue that I should be educated on?

Anyway I have locally changed things to use atomics and now rerunning things, etc.